Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -470,10 +470,9 @@ WarpXOpenPMDPlot::DumpToFile (WarpXParticleContainer* pc,
            }
 #else
            for (auto currDim = 0; currDim < AMREX_SPACEDIM; currDim++) {
-                std::shared_ptr< amrex::ParticleReal > curr(
-                    new amrex::ParticleReal[numParticleOnTile],
-                    [](amrex::ParticleReal const *p){ delete[] p; }
-                );
+                std::shared_ptr<amrex::ParticleReal> curr =
+                    std::make_unique<amrex::ParticleReal[]>(numParticleOnTile);
+
```

Comment:
This solution leaves me a bit unhappy. But we can't make it cleaner until C++20. If it is too misleading we might go back to the original version.